### PR TITLE
gives rig mod stations the ability to remove a user's rig modules when empty

### PR DESCRIPTION
**Pull request by Hatterhat**
[tweak]
~~Making it an intent thing is kinda fucked up on my part regardless but~~ it just checks if the user's rig has modules now

Empty (nothing in modules_to_install) spacesuit mod stations can be used (while standing inside them, of course) to uninstall modules from the wearer's suit. This only happens if your suit has modules to remove.
This takes the same time as it takes to install it, and puts the uninstalled module on the floor.

:cl:
 * rscadd: Rigsuit modification stations can now uninstall modules when empty. Use harm intent.

### CODE REVIEW

File: code/game/machinery/suit_modifier.dm [hunk 1]
```diff
@@ -225,6 +229,32 @@
...
+	activated = TRUE
+	use_power = 2
+	activation_animation()
+	var/obj/item/clothing/suit/space/rig/R = H.is_wearing_item(/obj/item/clothing/suit/space/rig, slot_wear_suit)
+	R.deactivate_suit()
+	if(R.modules.len)
+		// if we have something that's not a rig_module here we have a problem
+		var/obj/item/rig_module/RM = input(H, "Choose an upgrade to remove from [R].", R) as null|anything in R.modules
+		if(!RM|| !H.Adjacent(src) || H.incapacitated())
+			return
```
Comment (DamianX) [inline code/game/machinery/suit_modifier.dm:248]: you need to `unlock_atom(H)` before you return or the dude will be stuck in there

**Comment by DamianX** [comment]
I hate using intents for this kind of thing

File: code/game/machinery/suit_modifier.dm (hunk 1 above)
Comment (Hatterhat) [reply code/game/machinery/suit_modifier.dm:248]: whoops

**Comment by Hatterhat** [comment]
> 
> 
> I hate using intents for this kind of thing

it just checks for a worn rig and if it has modules now

**Comment by Kurfursten** [comment]
Update the intent page of the wiki.

I will remember this if you don't.

**Comment by Hatterhat** [comment]
> 
> 
> Update the intent page of the wiki.
> 
> I will remember this if you don't.

oh shit i forgot to update the changelog. yeah no it doesnt need harm intent

**Pull request merged**